File: Python/hangingout.py
'''
 *
 *	Problem: "Hanging Out on the Terrace"
 *	Problem ID: "hangingout"
 *	https://open.kattis.com/problems/hangingout
 *
 *	Solution: Python
'''

#-----number_prohibited-----#
def number_prohibited(limit, cases):
	curr = 0
	prohibited = 0
	
	for currCase in range(cases):
		line = input().split()
		action = line[0]
		number = int(line[1])
		
		if ( action == "enter" ):
		
			if (number + curr <= limit):
				curr += number
			else:
				prohibited += 1
				
		elif ( action == "leave" ):
			curr -= number
			
	return prohibited

#-----main-----#
line = input().split()
limit = int(line[0])
cases = int(line[1])

output = number_prohibited(limit, cases)
print(output)

File: Ruby/hangingout.rb
=begin
 *
 *	Problem:		Hanging Out on the Terrace
 *	Problem ID:	hangingout
 *	url:				https://open.kattis.com/problems/hangingout
 *	Solution:		Ruby
 *
=end

class Solver

	def solve
		limit, cases = gets.chomp.split(" ").map(&:to_i)
		
		current = 0
		prohibited = 0
	
		for this_case in 1..cases
			input = gets.chomp.split(" ")
			action = input[0]
			number = input[1].to_i
			
			if (action == "enter")
				if (number + current <= limit)
					current += number
				else
					prohibited += 1
				end
			elsif (action == "leave")
				current -= number
			end
		end
		
		puts(prohibited)
	end
	
end

solver = Solver.new
solver.solve
